Orange/widgets/visualize/tests/test_owmosaic.py

```diff
@@ -194,3 +194,23 @@ def test_does_not_crash(self):
         data = Table("housing.tab")
         widget.set_data(data)
         vizrank.toggle()
+
+    def test_zerodivision_error(self):
+        """
+        ZeroDivisionError should not be thrown when
+        a column with only NaN-s is loaded into widget.
+        GH-2046
+        """
+        table = Table(
+            Domain(
+                [ContinuousVariable("a"), ContinuousVariable("b"), ContinuousVariable("c")]),
+            np.array([
+                [0, np.NaN, 0],
+                [0, np.NaN, 0],
+                [0, np.NaN, 0]
+            ])
+        )
+        try:
+            self.send_signal("Data", table)
+        except ZeroDivisionError:
+            self.fail("Widget crashed due to ZeroDivisionError.")
```

Comment:
Using data with only NaNs should not result in any kind of error, not just ZeroDivision.
Why not make this test more general: rename from test_zerodivision_error to e.g. test_nan_column, remove the try, all exceptions will be shown when/if they happen.